src/middlewared/middlewared/plugins/failover_/boot.py

```diff
@@ -0,0 +1,58 @@
+from middlewared.schema import accepts, Bool, Dict, returns, Str
+from middlewared.service import private, Service
+
+
+class FailoverService(Service):
+
+    @private
+    async def retrieve_boot_ids(self):
+        return {
+            await self.middleware.call('failover.node'): await self.middleware.call('system.boot_id'),
+            await self.middleware.call(
+                'failover.call_remote', 'failover.node', [], {'raise_connect_error': False}
+            ): await self.middleware.call(
+                'failover.call_remote', 'system.boot_id', [], {'raise_connect_error': False}
+            ),
+        }
+
+    @accepts()
+    @returns(Dict(
+        Bool('reboot_required'),
+        Bool('node_a_reboot_required'),
+        Bool('node_b_reboot_required'),
+        Str('reason'),
+    ))
+    async def reboot_required(self):
+        """
+        Returns whether a reboot is required for failover/security system configuration changes to take effect.
+        """
+        return await self.check_reboot_required()
+
+    @private
+    async def check_reboot_required(self):
+        fips_change_info = await self.middleware.call('keyvalue.get', 'fips_toggled', False)
+        if not fips_change_info:
+            return {
+                'reboot_required': False,
+                'node_a_reboot_required': False,
+                'node_b_reboot_required': False,
+                'reason': 'No reboot required',
```

Comment:
We need to add the reasons for a reboot being required to `failover_/disabled_reasons.py::FailoverDisabledReasons` enum. When there isn't a reason a reboot is required, then please make `reason` NoneType. The UI already uses this information when an administrator logs in and adding it here will also make it show up in other utilities that our support team uses as well.